components/espcoredump/src/core_dump_flash.c

```diff
@@ -484,9 +486,21 @@ esp_err_t esp_core_dump_image_erase(void)
         return err;
     }
 
+    // helper to create (multiple of) 16 byte long write buffers
```

Comment:
There is a simpler way to do that I would prefer, it's to add at the beginning of this function the following snippet:
```
/* If flash is encrypted, we can only write blocks of 16 bytes. */
uint32_t helper[4] = { BLANK_COREDUMP_SIZE };
```
You can add the static assert right after (same as now, `sizeof(helper)` will return 16).